src/components/sidebar/types.ts

```diff
@@ -102,11 +103,21 @@ interface FilteredLinkNavItem extends EnrichedLinkNavItem {
 
 type FilteredNavItem = FilteredSubmenuNavItem | FilteredLinkNavItem
 
+export interface NavHighlightItem {
+	theme: ProductSlug
+	title: string
+	fullPath: string
+}
+
 /**
  *
  * For reference: this is also defined in react-components/docs-sidenav:
  * https://github.com/hashicorp/react-components/blob/main/packages/docs-sidenav/types.ts
  *
+ * TODO: it might be worth refactoring this to be a union of multiple types,
+ * one for each distinct type of menu item, rather than a single interface
+ * with optional properties to cover all possible menu item types.
+ *
```

Comment:
```suggestion
 * TODO: it might be worth refactoring this to be a union of multiple types,
 * one for each distinct type of menu item, rather than a single interface
 * with optional properties to cover all possible menu item types.
 * ref: https://app.asana.com/0/1202097197789424/1202405210286689/f
```